pkg/dao/external_resource.go

```diff
@@ -14,6 +17,39 @@ func GetExternalResourceDao() ExternalResourceDao {
 	return ExternalResourceDaoImpl{}
 }
 
+func (erd ExternalResourceDaoImpl) FetchGpgKey(url string) (string, error) {
+	timeout := 5 * time.Second
+	transport := http.Transport{ResponseHeaderTimeout: timeout}
+	client := http.Client{Transport: &transport, Timeout: timeout}
+
+	resp, clientError := client.Get(url)
+
+	// print(">>>>>>>------------", resp.StatusCode)
+	if clientError != nil {
+		return "", clientError
+	}
+
+	defer resp.Body.Close()
+
+	bodyBytes, err := io.ReadAll(resp.Body)
+
+	if err != nil {
+		return "", err
+	}
+
+	gpgKeyString := string(bodyBytes)
+
+	// print(">>>>>>>>>>>>>>>>>>>", gpgKeyString)
+
+	_, cryptoErr := crypto.NewKeyFromArmored(gpgKeyString)
```

Comment:
This will only load the first key.  It looks like there is a version of the old library that this library is still using (but presumably still maintained).  In your imports:

```
openpgp "github.com/ProtonMail/go-crypto/openpgp"
```
then here:
```
_, cryptoErr := openpgp.ReadArmoredKeyRing(strings.NewReader(gpgKeyString))
```